notification-server/src/bin/api.rs

```diff
@@ -54,8 +97,10 @@ async fn main() -> anyhow::Result<()> {
 
     tracing_subscriber::fmt::init();
 
-    let db_connection: Config = args.db_connection.parse()?;
-    let app_state = Arc::new(AppState { db_connection });
+    let app_state = Arc::new(AppState {
+        db_connection: DatabaseConnection::create(args.db_connection).await?,
+    });
```

Comment:
We used `deadpool` in our other projects so definitively a good option. Otherwise, this project was recommended as a good alternative but I haven't tried it out myself yet:
https://github.com/launchbadge/sqlx